src/utils/host_file_utils.cc

```diff
@@ -27,16 +28,29 @@
 
 void HostFileUtils::addPuttyKeyToOpenSshKeyFile( const QString& host, const QString& sshhostkeysFileName, const QString& sshKnownHostsFileName )
 {
+#ifdef Q_OS_WIN32
+    // Read the SshHostKey fingerprint form the registry.
+    QString rsaPrefix("0x23,0x");
+    QSettings puttySshHostKey("HKEY_CURRENT_USER\\Software\\SimonTatham\\PUTTY\\SshHostKeys", QSettings::NativeFormat);
```

Comment:
what happens if the specified registry does not exist yet because the user cleared it or putty never created it because it did not ran until at this point?